week3_ts/src/component/Tournament.tsx

```diff
@@ -0,0 +1,75 @@
+import { HandsomeGuy } from "@/assets/images";
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import styled from "styled-components";
+
+interface TournamentProps {
+    matchWinners: React.MutableRefObject<HandsomeGuy[]>;
+    fighterList: HandsomeGuy[];
+    setFighterList: React.Dispatch<React.SetStateAction<HandsomeGuy[]>>;
+}
+
+export default function Tournament(props: TournamentProps) {
+    const { matchWinners, fighterList, setFighterList } = props;
+
+    // 승자를 골랐을 때
+    const getSelectWinner = (pos: number) => {
+        matchWinners.current.push(fighterList[pos]);
+        setFighterList(fighterList.slice(2));
+    };
+
+    return (
+        <GameSection>
+            {fighterList.map((fighter, index) => {
+                if (index < 2) {
+                    return (
+                        <article onClick={() => getSelectWinner(index)} key={index}>
+                            <img src={fighter.url} />
+                            <div>{fighter.name}</div>
+                        </article>
+                    );
+                }
+            })}
```

Comment:
```suggestion
            {fighterList.slice(0, 2).map((fighter, index) => (
              <article onClick={() => getSelectWinner(index)} key={index}>
                <img src={fighter.url} />
                <div>{fighter.name}</div>
              </article>
            )}
```

이렇게도 축약할 수 있을 것 같네요!?